flink-ml-lib/src/main/java/org/apache/flink/ml/common/lossfunc/LossFunc.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.lossfunc;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.ml.common.feature.LabeledPointWithWeight;
+import org.apache.flink.ml.linalg.DenseVector;
+
+import java.io.Serializable;
+
+/**
+ * A loss function is to compute the loss and gradient with the given coefficient and training data.
+ */
+@Internal
+public interface LossFunc extends Serializable {
```

Comment:
Would it be better to merge `LossFunc` and its implementations into `optimizer` package? Would `LossFunc` be used in places other than `Optimizer` subclasses?

Reply:
`LossFunc` and `Optmizer` are two orthogonal concepts. For `LossFunc`, there could be logistic loss, least square loss, hinge loss, etc. For `Optimizer`, there could be sgd, adam, l-bfgs. 

Given a loss function, we can use different optimizers. Given a optimizer, we can optimize different loss functions.